lib/web/join_tokens.go

```diff
@@ -56,13 +57,40 @@ type scriptSettings struct {
 	appURI         string
 }
 
-func (h *Handler) createScriptJoinTokenHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
+// createTokenRequest is the expected request body of
+// the endpoint to create token
+type createTokenRequest struct {
+	Roles types.SystemRoles `json:"roles"`
+}
+
+func (h *Handler) createTokenHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
+	var req createTokenRequest
+	if err := httplib.ReadJSON(r, &req); err != nil {
+		log.WithError(err).Error("error reading body")
```

Comment:
Same comment as in the other backport, these errors should be logged by the caller.

Reply:
Thanks, removed on https://github.com/gravitational/teleport/pull/11256/commits/8679f545a0ea1a75f60801a5f3db717ad9b52760